arangod/Graph/Graph.cpp

```diff
@@ -239,6 +234,37 @@ void Graph::insertOrphanCollections(VPackSlice const arr) {
   }
 }
 
+std::set<std::string> Graph::vertexCollectionsUsedAsSatellites(
+    TRI_vocbase_t& vocbase) const {
+  std::set<std::string> list{};
+
+  std::shared_ptr<LogicalCollection> logicalCollection;
+  for (auto const& vertexCollection : vertexCollections()) {
+    logicalCollection =
+        graph::GraphManager::getCollectionByName(vocbase, vertexCollection);
+    if (logicalCollection != nullptr && logicalCollection->isSatellite()) {
+      list.emplace(vertexCollection);
+    }
+  }
+
+  return list;
+}
+
+std::string Graph::calculateMd5() const {
+  // Note: This implementation can be improved in the future.
+  // We can improve performance here, when we totally get rid of using
+  // the builder here.
+
+  // Creates VelocyPack graph representation
+  VPackBuilder md5Builder;
+  md5Builder.openObject();
+  generateMD5Representation(md5Builder);
+  md5Builder.close();
+
+  // Creates md5 checksum based on graph representation
+  return arangodb::rest::SslInterface::sslMD5(md5Builder.toString());
```

Comment:
I am not sure if it matters much,
but would we want to use `toJson()` here?
There subtle differences between toString() and toJson()